packages/catalog-client/src/CatalogClient.ts

```diff
@@ -28,13 +28,19 @@ import {
   CatalogEntitiesRequest,
   CatalogListResponse,
   DiscoveryApi,
+  IdentityApi,
 } from './types';
 
 export class CatalogClient implements CatalogApi {
   private readonly discoveryApi: DiscoveryApi;
+  private readonly identityApi: IdentityApi;
```

Comment:
So I think the best way to model this is to leave this client as is, but to add `token` options to all methods. Then we create a wrapper for that client that injects the token from the identity API in all requests. The wrapper code can even live in the frontend code as well, wherever we define the `catalogApiRef`.

Reply:
Token parameter added, wrapper client created in frontend